Source/TheLudoGameEngine/Classes/Token.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace TheLudoGameEngine
+{
+    public class Token
+    {
+        public int TokenID { get; set; }
+        public Player Player { get; set; }
+        public int GameBoardPosition { get; set; }
+        public string TokenColor { get; set; }
+        public int TokenNumber { get; set; }
+        public bool InNest { get; set; }
+        public bool InGoal { get; set; }
+
+        public int MoveToken(int dieResult)
+        {
+            for (int i = 1; i <= dieResult; i++)
+            {
+                if (this.GameBoardPosition >= 45)
+                {
+                    for (int y = i; y <= dieResult; y++)
+                    {
+                        i++;
+                        this.GameBoardPosition--;
+                    }
```

Comment:
här förstår jag inte riktigt vad som händer

Reply:
Kanske kan det delas i fler metoder

Reply:
Förstår helt och hållet att du/ni inte gör det haha. En av våra regler som vi kör på är att man behöver slå jämt att gå in i mål och denna variant var någon form av stegräkning och animations experiment vi hade i början, att loopen skulle fungera som en tick för att flytta pjäsen steg för steg både fram och tillbaka istället för att addera hela tärningens utgång direkt på pjäsens nuvarande position. Metoden är helt omgjord nu efter dagens revidering :)